src/awkward/_v2/contents/content.py

```diff
@@ -588,6 +588,72 @@ def min(self, axis=-1, mask=True, keepdims=False, initial=None):
     def max(self, axis=-1, mask=True, keepdims=False, initial=None):
         return self._reduce(awkward._v2._reducers.Max(initial), axis, mask, keepdims)
 
+    def argsort(self, axis=-1, ascending=True, stable=False, kind=None, order=None):
+        """
+        Args:
+            axis (int): this argument defines an axis along
+                which to sort indices. The default is `-1`, which
+                sorts indices along the innermost axis. Axes are defined similar to
+                Numpy axes. `axis=0` is the "first" axis and `axis=-1` is
+                the "last" axis. Unlike other operations, sorting does not
+                support `axis=None`.
+            ascending (bool): if `True`, the elements are sorted in ascending
+                order.
+            stable (bool): if `True`, the order of equivalent elements is guaranteed
+                to be preserved.
+            kind ({‘quicksort’, ‘mergesort’, ‘heapsort’, ‘stable’}, optional):
+                an optional sorting algorithm. This is similar to NumPy
+                sort.
+            order (str or list of str, optional): similar to NumPy.
+
+        Puts indices of Content elements in an ordered sequence: numeric or lexicographical,
+        ascending or descending.
+
+        This function is similar to NumPy [sort](https://numpy.org/doc/stable/reference/generated/numpy.argsort.html).
+        """
+        negaxis = -axis
+        branch, depth = self.branch_depth
+        if branch:
+            if negaxis <= 0:
+                raise ValueError(
+                    "cannot use non-negative axis on a nested list structure "
+                    "of variable depth (negative axis counts from the leaves "
+                    "of the tree; non-negative from the root)"
+                )
+            if negaxis > depth:
+                raise ValueError(
+                    "cannot use axis="
+                    + str(axis)
+                    + " on a nested list structure that splits into "
+                    "different depths, the minimum of which is depth="
+                    + str(depth)
+                    + " from the leaves"
+                )
+        else:
+            if negaxis <= 0:
+                negaxis = negaxis + depth
+            if not (0 < negaxis and negaxis <= depth):
+                raise ValueError(
+                    "axis="
+                    + str(axis)
+                    + " exceeds the depth of the nested list structure "
+                    "(which is " + str(depth) + ")"
+                )
```

Comment:
These kinds of things might be better as `.format(...)` than `+`. For anything but the most minimal cases, `.format(...)` is easier to read than `+`, and it makes conversions like `str(axis)` implicit. It's not reason enough to change this now, though.

Reply:
done